### PR TITLE
feat: support new fields for variables

### PR DESCRIPTION
**Pull request by massfords**
chore: bump asl-path-validator version
chore: npm audit fix
closes: #161 

This is the minimum to support the new variables fields

- add QueryLanguage field to root state machine
- add QueryLanguage field to states
- add new JSONata specific fields
- add some checks for invalid use of JSONPath fields with JSONata query language and vice versa

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
   "homepage": "https://github.com/ChristopheBougere/asl-validator#readme",
   "dependencies": {
     "ajv": "^8.12.0",
-    "asl-path-validator": "^0.14.2",
+    "asl-path-validator": "^0.15.0",
```
Comment (massfords) [inline package.json:37]: new version supports variables in JSONPath expressions and also adds a new format type for `ResultPath` fields which are `ReferencePath` expressions but cannot contain variables.

File: src/schemas/choice.json [hunk 2]
```diff
@@ -139,6 +139,9 @@
         },
         "IsTimestamp": {
           "type": "boolean"
+        },
+        "Condition": {
+          "$ref": "jsonata.json#/definitions/condition"
```
Comment (massfords) [inline src/schemas/choice.json:144]: >In a JSONata Choice State, each Choice Rule MUST have a "Condition" field. The "Condition" field accepts a boolean value or a JSONata string that must evaluate to a boolean value.

File: src/schemas/map.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
       "$ref": "paths.json#/definitions/asl_path"
     },
     "ResultPath": {
-      "$ref": "paths.json#/definitions/asl_ref_path"
+      "$ref": "paths.json#/definitions/asl_result_path"
```
Comment (massfords) [inline src/schemas/map.json:26]: >In a JSONPath state, the "ResultPath" field MUST NOT reference a variable.

I added a new format type in asl-path-validator to support this

File: src/schemas/state-machine.json [hunk 4]
```diff
@@ -24,6 +24,9 @@
     "TimeoutSeconds": {
       "type": "integer",
       "minimum": 0
+    },
+    "QueryLanguage": {
+      "$ref": "jsonata.json#/definitions/queryLanguage"
```
Comment (massfords) [inline src/schemas/state-machine.json:29]: >A state machine specifies its query language in the [top-level "QueryLanguage" field](https://states-language.net/#toplevelfields), which overrides the default and sets the query language for every state in the state machine.

**Comment by massfords** [comment]
@ChristopheBougere I'm still not sure what I'm doing wrong with these commit messages. I've got the latest repo, ran install, and did the build before making any changes. My initial commit had a leading blank line but I suspect Webstorm trimmed it. I did a `git commit --amend` from the command line and added the leading blank line and then force pushed to my branch.

Anyway, if you can take a look and fix it would be appreciated. I think this is a good first pass at supporting the new variables/JSONata feature. There's a fair amount of additional work with respect to validating the JSONata expressions and ensuring that the referenced variables are in scope. There's also other validation like checking the length and naming rules for variables introduced in `Assign` activities.  Perhaps additional work for later.

**Review by ChristopheBougere — APPROVED** [review]
Looking good!

**Comment by ChristopheBougere** [comment]
> @ChristopheBougere I'm still not sure what I'm doing wrong with these commit messages. I've got the latest repo, ran install, and did the build before making any changes. My initial commit had a leading blank line but I suspect Webstorm trimmed it. I did a `git commit --amend` from the command line and added the leading blank line and then force pushed to my branch.
> 
> Anyway, if you can take a look and fix it would be appreciated. I think this is a good first pass at supporting the new variables/JSONata feature. There's a fair amount of additional work with respect to validating the JSONata expressions and ensuring that the referenced variables are in scope. There's also other validation like checking the length and naming rules for variables introduced in `Assign` activities. Perhaps additional work for later.

Sure I will amend the commit message
It seems you just miss a blank line between the title and the body
image

It might be because you squashed your commits and the new commit message wasn't well formatted, or something like this

**Pull request merged**